File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/LastMatchedReciprocalFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

import java.util.OptionalDouble;

@Component
public class LastMatchedReciprocalFunction implements ProgressFunction {

    @Override
    public OptionalDouble apply(SettledBet bet) {
        return bet.getBetAction().getMarketPrices().getLastMatchedReciprocal();
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/LastMatchedReciprocalFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.SettledBet
import org.springframework.stereotype.Component

@Component
class LastMatchedReciprocalFunction : ProgressFunction {

    override fun invoke(bet: SettledBet): Double? {
        val reciprocal = bet.betAction.marketPrices.lastMatchedReciprocal
        return if (reciprocal.isPresent) reciprocal.asDouble else null
    }

}
